src/writer.js

```diff
@@ -7,32 +7,55 @@ const encode = require('./encode')
 const tracerVersion = require('../lib/version')
 
 class Writer {
-  constructor (url, size) {
+  constructor (url, size, processors) {
     this._queue = []
     this._url = url
     this._size = size
+    this._processors = processors
   }
 
   get length () {
     return this._queue.length
   }
 
+  // A trace is considered filtered out if it is undefined, or if it contains no spans
+  isTraceFilteredOut (trace) {
+    return trace === undefined || trace === []
+  }
+
+  applyProcessors (trace) {
```

Comment:
I would probably extract this to an external `Pipeline` class or at the very least an external `process()` function similar to how `format()` and `encode()` are external.

Reply:
Nitpick, but I think `run` is probably more semantically correct for a pipeline.